tests/Resources/ti.ui.window.test.js

```diff
@@ -683,6 +678,85 @@ describe('Titanium.UI.Window', function () {
 			should(win.custom).eql(1234);
 		});
 
+		describe('#close()', () => {
+			it('is a Function', () => {
+				win = Ti.UI.createWindow({
+					backgroundColor: '#0000ff'
+				});
+
+				should(win).have.a.property('close').which.is.a.Function();
+			});
+
+			it('returns a Promise', finish => {
+				win = Ti.UI.createWindow({
+					backgroundColor: '#0000ff'
+				});
+
+				const openPromise = win.open();
+				openPromise.then(() => {
+					const result = win.close();
+					result.should.be.a.Promise();
+					result.then(() => finish()).catch(e => finish(e));
+				}).catch(e => finish(e));
+			});
+
+			it('called on unopened Window rejects Promise', finish => {
+				win = Ti.UI.createWindow({
+					backgroundColor: '#0000ff'
+				});
+
+				const result = win.close();
+				result.should.be.a.Promise();
+				result.then(() => finish(new Error('Expected #close() to be rejected on unopened Window'))).catch(_e => finish());
+			});
+
+			it('called twice on Window rejects second Promise', finish => {
+				win = Ti.UI.createWindow({
+					backgroundColor: '#0000ff'
+				});
+
+				win.open().then(() => {
+					win.close().then(() => {
+						win.close().then(() => finish(new Error('Expected second #close() call on Window to be rejected'))).catch(e => finish());
+					}).catch(e => finish(e));
+				}).catch(e => finish(e));
+			});
+		});
+
+		describe('#open()', () => {
+			it('is a Function', () => {
+				win = Ti.UI.createWindow({
+					backgroundColor: '#0000ff'
+				});
+
+				should(win).have.a.property('open').which.is.a.Function();
+			});
+
+			it('returns a Promise', finish => {
+				win = Ti.UI.createWindow({
+					backgroundColor: '#0000ff'
+				});
+
+				const result = win.open();
+				result.should.be.a.Promise();
+				result.then(() => finish()).catch(e => finish(e));
+			});
+
+			it('called twice on same Window rejects second Promise', finish => {
+				win = Ti.UI.createWindow({
+					backgroundColor: '#0000ff'
+				});
+
+				const first = win.open();
+				first.should.be.a.Promise();
+				first.then(() => {
```

Comment:
<!--
  0 failure: 
  1 warning:  tests/Resources/t...
  
  
  DangerID: danger-id-Danger;
  File: tests/Resources/ti.ui.window.test.js;
  Line: 752;
-->

- :warning: tests/Resources/ti.ui.window.test.js line 752 – Each then() should return a value or throw (promise/always-return)